contracts/ERC721A.sol

```diff
@@ -270,7 +324,7 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
      * Tokens start existing when they are minted (`_mint`),
      */
     function _exists(uint256 tokenId) internal view returns (bool) {
-        return tokenId < currentIndex;
+        return tokenId < currentIndex && !_ownerships[tokenId].burned;
```

Comment:
This is the most straightforward way to introduce burn, but I have concerns over the additional SLOAD

_exists is primarily used in three critical/write places:
1. ownershipOf
2. mint
3. transfer

- ownershipOf performs SLOADs later, so we can presumably batch these to determine whether burned or not
- mint presumably always mints serially and we shouldnt need to check for `burned`
- calls ownershipOf

my thoughts are it makes more sense to perform burn logic in ownershipOf

Reply:
We can manually inline the _exists for ownershipOf.

Around 200 gas saved from avoiding a [warm SLOAD](https://eips.ethereum.org/EIPS/eip-2929). 

**Before:**
https://github.com/Vectorized/ERC721A/blob/33d4b4115b3018e8689f19adfdb03c582cd6d368/contracts/ERC721A.sol#L187
```
·-----------------------------------------------------------|---------------------------|-------------|
|                   Solc version: 0.8.11                    ·  Optimizer enabled: true  ·  Runs: 800  ·
····························································|···························|·············|
|  Methods                                                  ·              100 gwei/gas               ·
······································|·····················|·············|·············|·············|
|  Contract                           ·  Method             ·  Min        ·  Max        ·  Avg        ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  burn               ·      49666  ·     118934  ·      97307  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  initOneIndexed     ·          -  ·          -  ·      87709  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  safeMint           ·      98764  ·     115864  ·     106811  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  burn               ·      49666  ·      90310  ·      74260  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  initOneIndexed     ·          -  ·          -  ·      87709  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  safeMint           ·      77511  ·      94611  ·      82716  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  setOwnersExplicit  ·      82283  ·      82480  ·      82382  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  approve            ·          -  ·          -  ·      51137  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  mint               ·      92806  ·     101402  ·      99110  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeMint           ·      79870  ·     125334  ·      89917  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeTransferFrom   ·          -  ·          -  ·      94319  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  setApprovalForAll  ·          -  ·          -  ·      46219  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  transferFrom       ·          -  ·          -  ·      87218  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  safeMint           ·      79858  ·      94611  ·      85566  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  setOwnersExplicit  ·      48787  ·     126038  ·      86045  ·
······································|·····················|·············|·············|·············|
```

**After:**
https://github.com/Vectorized/ERC721A/blob/a2ef813f001b913c253577e830435b7549c6e6a0/contracts/ERC721A.sol#L174
```
·-----------------------------------------------------------|---------------------------|-------------|
|                   Solc version: 0.8.11                    ·  Optimizer enabled: true  ·  Runs: 800  ·
····························································|···························|·············|
|  Methods                                                  ·              100 gwei/gas               ·
······································|·····················|·············|·············|·············|
|  Contract                           ·  Method             ·  Min        ·  Max        ·  Avg        ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  burn               ·      49240  ·     119028  ·      97262  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  initOneIndexed     ·          -  ·          -  ·      87709  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  safeMint           ·      98764  ·     115864  ·     106811  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  burn               ·      49240  ·      89996  ·      73890  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  initOneIndexed     ·          -  ·          -  ·      87709  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  safeMint           ·      77511  ·      94611  ·      82716  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  setOwnersExplicit  ·      82126  ·      82323  ·      82225  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  approve            ·          -  ·          -  ·      50924  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  mint               ·      92806  ·     101402  ·      99110  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeMint           ·      79870  ·     125334  ·      89917  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeTransferFrom   ·          -  ·          -  ·      94106  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  setApprovalForAll  ·          -  ·          -  ·      46219  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  transferFrom       ·          -  ·          -  ·      87005  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  safeMint           ·      79858  ·      94611  ·      85566  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  setOwnersExplicit  ·      48787  ·     125567  ·      85790  ·
······································|·····················|·············|·············|·············|
```